sycl/include/sycl/ext/oneapi/experimental/group_sort.hpp

```diff
@@ -124,16 +126,98 @@ template <typename Group, typename Iter, typename Compare, size_t Extent>
 std::enable_if_t<!detail::is_sorter<Compare, Group, Iter>::value, void>
 joint_sort(experimental::group_with_scratchpad<Group, Extent> exec, Iter first,
            Iter last, Compare comp) {
-  joint_sort(exec.get_group(), first, last,
-             experimental::default_sorter<Compare>(exec.get_memory(), comp));
+  joint_sort(
+      exec.get_group(), first, last,
+      //  experimental::default_sorter<Compare>(exec.get_memory(), comp));
+      default_sorters::joint_sorter<Compare>(exec.get_memory(), comp));
 }
 
 template <typename Group, typename Iter, size_t Extent>
 std::enable_if_t<sycl::is_group_v<std::decay_t<Group>>, void>
 joint_sort(experimental::group_with_scratchpad<Group, Extent> exec, Iter first,
            Iter last) {
   joint_sort(exec.get_group(), first, last,
-             experimental::default_sorter<>(exec.get_memory()));
+             //  experimental::default_sorter<>(exec.get_memory()));
+             default_sorters::joint_sorter<>(exec.get_memory()));
+}
+
+template <typename Group, typename T, typename U, typename Sorter>
+// TODO: ADD check for is_sorter , detail::is_sorter<Sorter, Group,
+// Iter>::value, void>::type>
+std::tuple<T, U> sort_key_value_over_group(Group g, T key, U value,
+                                           Sorter sorter) {
+#ifdef __SYCL_DEVICE_ONLY__
+  return sorter(g, key, value);
+#else
+  (void)g;
+  (void)key;
+  (void)value;
+  (void)sorter;
+  throw sycl::exception(
+      std::error_code(PI_ERROR_INVALID_DEVICE, sycl::sycl_category()),
+      "Group algorithms are not supported on host device.");
+#endif
+}
+
+template <typename GroupHelper, typename T, typename U, typename Compare,
+          typename Group, std::size_t Extent>
+std::tuple<T, U> sort_key_value_over_group(
+    experimental::group_with_scratchpad<Group, Extent> exec, T key, U value,
+    Compare comp) {
+  return sort_key_value_over_group(
+      exec.get_group(), key, value,
+      default_sorters::group_key_value_sorter<T, U, Compare>(exec.get_memory(),
+                                                             comp));
+}
+
+template <typename T, typename U, typename Group, std::size_t Extent>
+std::tuple<T, U> sort_key_value_over_group(
+    experimental::group_with_scratchpad<Group, Extent> exec, T key, U value) {
+  return sort_key_value_over_group(
+      exec.get_group(), key, value,
+      default_sorters::group_key_value_sorter<T, U>(exec.get_memory()));
+}
+
+template <typename Group, typename T, typename U, std::size_t Extent,
+          std::size_t ElementsPerWorkItem, typename Compare,
+          typename Property = detail::is_blocked>
```

Comment:
According to the [document ](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc) here should be `typename Properties = ext::oneapi::experimental::empty_properties_t>`.

Reply:
Good point! Fixed. Thanks!